src/Tools/dotnet-trace/CommandLine/Commands/CollectCommand.cs

```diff
@@ -194,11 +194,21 @@ private static async Task<int> Collect(CancellationToken ct, IConsole console, i
                                         if (hasConsole)
                                         {
                                             lineToClear = Console.CursorTop - 1;
+
+                                            // in case of INPUT redirection the vertical position does not seem to be updated
+                                            // as expected. So it is needed to count the lines to clear: here only 2 lines are
+                                            // displayed via Console.Out.WriteLine (time + size and keys to exit).
+                                            // If new lines are added/removed in the future, the number will need to be updated accordingly.
+                                            if (!vTermMode.IsEnabled) lineToClear -= 1;  // 2 lines - 1 from lineToClear initialization = 1
                                             ResetCurrentConsoleLine(vTermMode.IsEnabled);
+
+                                            // don't output anything if there is no console
+                                            // otherwise these lines might appear both in case of OUTPUT > file redirection
+                                            // and & background in Linux
```

Comment:
I'm not sure I follow this comment.  Wouldn't writing to standard out follow the redirection regardless of whether the process is run in the background with `&`?  Is this some oddity you observed with the `System.Console` API?

Reply:
I don't know how it works underneath but I've seen these two lines appearing in the console with & but also if you redirect to a file (with > output.txt for example) and even if input is redirected with < input.txt!
![image](https://user-images.githubusercontent.com/10163781/87597678-7ecd7200-c6f2-11ea-8761-4a9897dabf90.png)

Reply:
If you put a command in the background with `&` but don't redirect the output with `>`, it usually will print to the console unless the app is designed to pick up that it was run in the background.  Redirecting input with `<` similarly won't change the output fd of the application.  If we want to make a conscious decision to change output on _any_ redirection or background status, I think we should change the concept of `hasConsole` to be checking those things directly.